hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -2968,6 +2968,11 @@ public ListKeysLightResult listKeysLight(String volumeName,
   public SnapshotInfo getSnapshotInfo(String volumeName, String bucketName,
                                       String snapshotName) throws IOException {
     metrics.incNumSnapshotInfos();
+    // Updating the volumeName & bucketName in case the bucket is a linked bucket. We need to do this before a
+    // permission check, since linked bucket permissions and source bucket permissions could be different.
+    ResolvedBucket resolvedBucket = resolveBucketLink(Pair.of(volumeName, bucketName), false);
+    volumeName = resolvedBucket.realVolume();
```

Comment:
nit: we should not update the `volumeName` and `bucketName` pass to the method. Just rather pass the resolved bucket and volume to the downstream method.

Reply:
done